client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/list-item.js

```diff
@@ -0,0 +1,151 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+import { trim } from 'lodash';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import CarrierIcon from '../../components/carrier-icon';
+import Dialog from 'components/dialog';
+import {
+	getCarrierAccountsState,
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/selectors';
+import {
+	setVisibilityDisconnectCarrierDialog,
+	disconnectCarrier,
+} from 'woocommerce/woocommerce-services/state/carrier-accounts/actions';
+
+export const CarrierAccountListItem = ( props ) => {
+	const {
+		data,
+		translate,
+		siteId,
+		showDisconnectDialog,
+	} = props;
+
+	const renderIcon = carrierId => {
+		return <div className="carrier-accounts__list-item-carrier-icon">
+			<CarrierIcon carrier={ carrierId } size={ 18 } />
+		</div>;
+	};
+
+	const renderName = name => {
+		const carrierName = name && '' !== trim( name ) ? name : translate( 'Untitled' );
+		return <div className="carrier-accounts__list-item-name">
+			<span>{ carrierName }</span>
+		</div>;
+	};
+
+	const renderCredentials = credentials => {
+		return <div className="carrier-accounts__list-item-credentials">
+			<span>{ credentials }</span>
+		</div>;
+	}
+
+	const showDisconnectDialogHandler = () => {
+		props.setVisibilityDisconnectCarrierDialog( siteId, data.carrierId, true );
+	}
+
+	const hideDisconnectDialogHandler = () => {
+		props.setVisibilityDisconnectCarrierDialog( siteId, data.carrierId, false );
+	}
+
+	const connectCarrierHandler = () => {
+		const url = new URL( window.location.href );
+		url.searchParams.set( 'carrier', data.carrierId );
+		window.location.href = url.href;
+	}
+
+	const renderActions = ( credentials ) => {
+		const connectButton = () => {
+			return <Button compact onClick={ connectCarrierHandler }>
```

Comment:
If we have button style with `<a>`, we can then avoid the onclick handler

Reply:
Yeah, better that way indeed.